src/txmempool.cpp

```diff
@@ -447,6 +450,15 @@ void CTxMemPool::removeUnchecked(txiter it)
     BOOST_FOREACH(const CTxIn& txin, it->GetTx().vin)
         mapNextTx.erase(txin.prevout);
 
+    if (vTxHashes.size() > 1) {
+        vTxHashes[it->vTxHashesIdx] = vTxHashes[vTxHashes.size() - 1];
+        txiter newhashit = mapTx.find(vTxHashes[it->vTxHashesIdx].first);
```

Comment:
Yes, I found that out after looking it up :)

On June 16, 2016 9:19:08 PM PDT, Pieter Wuille <notifications@github.com> wrote:
>> @@ -447,6 +450,15 @@ void CTxMemPool::removeUnchecked(txiter it)
>>      BOOST_FOREACH(const CTxIn& txin, it->GetTx().vin)
>>          mapNextTx.erase(txin.prevout);
>>  
>> +    if (vTxHashes.size() > 1) {
>> +        vTxHashes[it->vTxHashesIdx] = vTxHashes[vTxHashes.size() -
>1];
>> +        txiter newhashit =
>mapTx.find(vTxHashes[it->vTxHashesIdx].first);
>
>They are never invalidated until you delete the item itself.
>http://www.boost.org/doc/libs/1_60_0/libs/multi_index/doc/tutorial/indices.html#guarantees
>
>
>---
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bitcoin/pull/8068/files/ba9b1415c5c226eee55f77b10095335240d7af5e..52783001e3cb06759bc823c71ab04e3690b81ad1#r67459055